### ISSUE TITLE
Fix S2325 FP: Primary Constructor Support

### ISSUE THREAD

**Issue by Noel-Voelin** [SonarSource/sonar-dotnet#8567]
### Description

S2325 flags methods in classes with a primary constructor when a private method references fields set in that constructor. The suggested fix is to make the private method static. However, this is inaccurate because the fields injected by the primary constructor are instance-specific, and attempting to make these methods static results in a compiler error.

### Repro steps

Example where the false flagging happens:
```C#
public class UtilitiesService(NumberSumService numberSumService) : IUtilitiesService
{
    public int Sum(int a, int b)
    {
        return GetResultsFromServiceCall(a, b);
    }

    private int GetResultsFromServiceCall(int a, int b)
    {
        return numberSumService.SumNumbers(a, b);
    }
}
```

### Known workarounds
Known workaround would be to continue using explicit constructors.
```C#
public class UtilitiesService : IUtilitiesService
{
    private readonly NumberSumService numberSumService;

    public UtilitiesService(NumberSumService numberSumService)
    {
        this.numberSumService = numberSumService;
    }

    public int Sum(int a, int b)
    {
        return GetResultsFromServiceCall(a, b);
    }

    private int GetResultsFromServiceCall(int a, int b)
    {
        return numberSumService.SumNumbers(a, b);
    }
}
```
### Expected behavior

[S2325](https://sonarsource.github.io/rspec/#/rspec/S2325/csharp) is not triggered on methods that reference a field injected by the Primary Constructor (Behave the same as with explicit Constructor)

### Actual behavior
[S2325](https://sonarsource.github.io/rspec/#/rspec/S2325/csharp) is triggered.

### Related information
* C#/VB.NET Plugins version: 9.17.0.82934
* Rider Version: 2023.3.2
* MSBuild / dotnet version: 14.3.0 / 8.0.100
* SonarScanner for .NET version (if used)
* Operating System: Windows 11 Pro 22H2 22621.3007

**Comment by sebastien-marichal** [comment]
Hello @Noel-Voelin,

Thank you for reporting this issue.

Unfortunately, I am not able to reproduce it.
This is what I have:

```cs
public interface IUtilitiesService
{
    int Sum(int a, int b);
}

public class NumberSumService
{
    public int SumNumbers(int a, int b)
    {
        return a + b;
    }
}

public class UtilitiesService(NumberSumService numberSumService) : IUtilitiesService
{
    public int Sum(int a, int b)
    {
        return GetResultsFromServiceCall(a, b);
    }

    private int GetResultsFromServiceCall(int a, int b)
    {
        return numberSumService.SumNumbers(a, b);
    }
}
```

In my case, S2325 is not raising on `GetResultsFromServiceCall`.

I see you mention your version of MSBuild is 14.3.0? It seems pretty old, is the information correct?
In which context did you get this issue? In your IDE or when invoking SonarScanner for .NET (locally or in the CI)?

Thank you,

**Comment by gregory-paidis-sonarsource** [comment]
Hey @Noel-Voelin ,
I managed to reproduce it.
I will add it to our backlog and it will most likely be fixed soon.

Thanks for raising this!

**Comment by Noel-Voelin** [comment]
@sebastien-marichal Sorry for not responding any quicker! About the msbuild version, I am quite frankly not sure. The error occurred to me in local analysis as well as CI/CD executions tho! But I see that @gregory-paidis-sonarsource was able to reproduce the issue!

Thanks so much for your efforts guys!

**Comment by antonioaversa** [comment]
@gregory-paidis-sonarsource I don't currently have the bandwidth to deal with this PR. Can you please find another reviewer?
Thanks and apologies for the delay.
